src/test/operators/table_scan_sorted_test.cpp

```diff
@@ -0,0 +1,175 @@
+#include "base_test.hpp"
+#include "gtest/gtest.h"
+#include "operators/limit.hpp"
+#include "operators/table_wrapper.hpp"
+
+namespace opossum {
+
+using Params = std::tuple<EncodingType, bool, std::pair<PredicateCondition, std::vector<AllTypeVariant>>, DataType,
+                          OrderByMode, bool>;
+
+class OperatorsTableScanSortedTest : public BaseTest, public ::testing::WithParamInterface<Params> {
+ protected:
+  void SetUp() override {
+    _encoding_type = std::get<0>(GetParam());
+    _data_type = std::get<3>(GetParam());
+    _order_by = std::get<4>(GetParam());
+    _expected = std::get<2>(GetParam()).second;
+    const bool nullable = std::get<5>(GetParam());
+    const bool use_reference_segment = std::get<1>(GetParam());
+
+    const bool ascending = _order_by == OrderByMode::Ascending || _order_by == OrderByMode::AscendingNullsLast;
+    const bool nulls_first = _order_by == OrderByMode::Ascending || _order_by == OrderByMode::Descending;
+
+    if (!ascending) {
+      std::reverse(_expected.begin(), _expected.end());
+    }
+
+    _table_column_definitions.emplace_back("a", _data_type, nullable);
+
+    auto table = Table::create_dummy_table(_table_column_definitions);
+
+    if (nullable && nulls_first) {
+      table->append({NULL_VALUE});
+      table->append({NULL_VALUE});
+      table->append({NULL_VALUE});
+    }
+
+    const int table_size = 10;
```

Comment:
Would it be possible to just use a table in a normal `.tbl` file. Possibly with one column per configuration? tbl files are much easier to follow than code that generates a table base on a plethora of parameters. And only if we can follow the code/table we can actually say whether the test covers all it should cover or not.

Reply:
It would certainly be possible but not as a single `.tbl` file because the initial tables for our tests have a different number of rows (see below). As far as I understand it, the importer can't handle `.tbl`files where some fields are empty, which is why we would have to split them.
We are currently still discussing if reference segments don't need to be tested (as mentioned above) and if this is the case we would need 2 different files otherwise 4.
If this fine then we could implement it this way.

| asc | asc nulls first | asc reference | asc reference nulls first |
| --- | --- | --- | --- |
| 0   | NULL           | 0      | NULL               |
| 1   | NULL           | 0      | NULL               |
| 2   | NULL           | 1      | NULL               |
| 3   | 0              | 1      | 0                  |
| 4   | 1              | 2      | 0                  |
| 5   | 2              | 2      | 1                  |
| 6   | 3              | 3      | 1                  |
| 7   | 4              | 3      | 2                  |
| 8   | 5              | 4      | 2                  |
| 9   | 6              | 4      | 3                  |
|     | 7              | 5      | 3                  |
|     | 8              | 5      | 4                  |
|     | 9              | 6      | 4                  |
|     |                | 6      | 5                  |
|     |                | 7      | 5                  |
|     |                | 7      | 6                  |
|     |                | 8      | 6                  |
|     |                | 8      | 7                  |
|     |                | 9      | 7                  |
|     |                | 9      | 8                  |
|     |                |        | 8                  |
|     |                |        | 9                  |
|     |                |        | 9                  |

Reply:
> because the initial tables for our tests have different numbers of rows (see below).

Multiple tables then, no problemo. What is important to me is that the input data (+ operations and expected output)  to your tests is decipherable, and currently that requires some real work on the readers part.